### DISCUSSION TITLE
Finding the last time `brew update` was ran (perhaps for a tap)

### DISCUSSION THREAD

**Discussion by nemith** [Homebrew/discussions#3257] [Tap maintenance and Homebrew development]
I am trying to come up with a strategy for deploying some internal tools and homebrew + a private tap seems like a good idea, however I want my CLI tools to inform the user when they are outdated. 

I can run `brew outdated <tap/formula> --json` pretty easily but that assumes the user has ran `brew update` recently.  It would be nice to force a `brew update` on behalf of the user if it hasn't been updated in some time frame like 24hrs.    I don't want to run `brew update` on every call. 

The push here is really to cut down on support tickets coming into my (small) team and making sure users are using the latest internal tool binaries on their system.   

Also open to other options for doing an out-of-date version check.

**Comment by svnk** [comment]
as 'brew update' fetches the state of the git repo of the brew formulas, we can check, when git in that repo was last run by e.g. executing the following:
`date -r /opt/homebrew/.git/HEAD`

**Comment by konnano** [comment]
Implicit execution is not supported, but you can bypass the check every time.
```
echo '
export HOMEBREW_NO_AUTO_UPDATE=1
brew(){
 TI=$(date +%s)
 DA=$(command brew --repo)/.git/HEAD
 LS=$(date -r $DA "+%Y-%m-%d %H:%M:%S" 2>/dev/null)||${die?error}
 if [ $(uname) = Darwin ];then
  LS1=$(( $(date -jf "%Y-%m-%d %H:%M:%S" "$LS" +%s 2>/dev/null)+86400 ))
   if [ $TI -gt $LS1 ];then
    touch $DA;
    command brew update
   fi
  command brew "$@"
 else
  LS1=$(( $(date +%s --date "$LS" 2>/dev/null)+86400 ))
   if [ $TI -gt $LS1 ];then
    touch $DA
    command brew update
   fi
  command brew "$@"
 fi
}' >> ~/$(echo .${SHELL##*/}rc)
```
